devtools/x/src/cargo.rs

```diff
@@ -437,3 +392,153 @@ impl<'a> CargoCommand<'a> {
         }
     }
 }
+
+/// Arguments for the Cargo package selector.
+#[derive(Debug, StructOpt)]
+pub struct CargoPackageArgs {
+    #[structopt(long, short, number_of_values = 1)]
+    /// Run on the provided packages
+    pub(crate) package: Vec<String>,
+    #[structopt(long, short)]
+    /// Run on packages changed since the merge base of this commit
+    changed_since: Option<String>,
+    #[structopt(long)]
+    /// Run on all packages in the workspace
```

Comment:
It's taking me a bit longer to grok this since CargoPackages could represent the entire workspace, which seems a pretty large departure from where we were.   Not bad, just different.   I think this section needs more doc.

Reply:
naming.....   maybe something like "CommandTargets" instead of CargoPackages.....

Reply:
TODO: document design requirements and decisions here.

Reply:
Renamed it to `SelectedPackages`.